pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -652,6 +653,10 @@ func (s *Reconciler) createVirtualMachine(ctx context.Context, nicName string) e
 			return errors.Errorf("MachineConfig managedIdentity is missing on machine %s", s.scope.Machine.Name)
 		}
 
+		if s.scope.Machine.Labels == nil || s.scope.Machine.Labels[machinev1.MachineClusterIDLabel] == "" {
+			return errors.Errorf("machine is missing %q label", machinev1.MachineClusterIDLabel)
```

Comment:
Add which machine (i.e., %q... machine.Name)

Reply:
The machine name is added at reconciler.go:124

```go
if err := s.createVirtualMachine(ctx, nicName); err != nil {
	return errors.Wrapf(err, "failed to create vm %s ", s.scope.Machine.Name)
}
```